main.tf

```
@@ -0,0 +1,101 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+  region = "us-east-1"
+}
+
+# Create a VPC
+resource "aws_vpc" "example" {
+  cidr_block = "10.0.0.0/16"
+}
+
+# Define local var for public and private cidr
+locals {
+  public_cidr  = ["10.0.0.0/24", "10.0.1.0/24"]
+  private_cidr = ["10.0.2.0/24", "10.0.3.0/24"]
+}
+
+# Configure two public subnet
+resource "aws_subnet" "public" {
+  count      = 2
```

This will only work if we want to deploy 2 subnets.
If we want to deploy more than 2 or be able to specify arbitrary number of subnets - it's better to use this construct
```suggestion
  count      = length(local.public_cidr)
```
What it will do is count how many subnets you have specified in the list and return the value
More information about this here:
https://www.terraform.io/language/functions/length

---

added